lib/OpenQA/Schema/Result/Comments.pm

```diff
@@ -132,9 +132,18 @@ Returns C<build_nr>, C<type> and optionally C<description> if C<$self> is tag,
 e.g. 'tag:0123:important:GM' returns a list of '0123', 'important' and 'GM'.
 =cut
 sub tag ($self) {
-    $self->text
-      =~ /\btag:((?<version>[-.@\d\w]+)-)?(?<build>[-.@\d\w]+):(?<type>[-@\d\w]+)(:(?<description>[-.@\d\w]+))?\b/;
-    return $+{build}, $+{type}, $+{description}, $+{version};
+    my $text = $self->text;
+    if (
+        (
+            $text
+            =~ /\btag:((?<version>[-.@\d\w]+)-)?(?<build>[-.@\d\w]+):(?<type>[-@\d\w]+)(:(?<description>[-.@\d\w]+))?\b/
+        )
+        || ($text
+            =~ /\btag:"((?<version>[-.@\d\w]+)-)?(?<build>[-.@\d\w\s\+:]+)":(?<type>[-@\d\w]+)(:(?<description>[-.@\d\w]+))?\b/
+        ))
+    {
```

Comment:
I think having two nearly identical variants of regex strings which already contain optional parts are not that readable. I suggest to combine both. Something like

```suggestion
    if ($text =~ /\btag:(((?<version>[-.@\d\w]+)-)?(?<build>[-.@\d\w]+)|"((?<version>[-.@\d\w]+)-)?(?<build>[-.@\d\w\s\+:]+)"):(?<type>[-@\d\w]+)(:(?<description>[-.@\d\w]+))?\b/) {
```

This is of course also nearly unreadable so I suggest to invest some effort in improving, e.g. with variables.

Also the commit message could have a bit more details, e.g. what issue is fixed, why this needs to change, to which versions of which product it applies, link to a bug or a fate entry, the choices you made, etc. Also see https://commit.style/ or http://chris.beams.io/posts/git-commit/ as a helpful guide how to write good commit messages.

Reply:
I don't think making one huge regex out of it helps with readability.

Reply:
Then variables in the string. But if/else with *two* huge regex strings isn't better

Reply:
@okurz in perl re is possible to have two subpatterns with the same name? in that case one ugly regex is still better than two :D